src/modules/Footer.jsx

```diff
@@ -0,0 +1,269 @@
+  };
+
+  const moveUp = () => {
+    const newBoard = new Array(squareBoardSize);
+    for (let i = 0; i < boardSize; i++) {
+      updateNewBoardVertical(newBoard, i);
+    }
+    setNewBoard(newBoard);
+  };
+
+  const moveDown = () => {
+    const newBoard = new Array(squareBoardSize);
+    for (let i = 0; i < boardSize; i++) {
+      updateNewBoardVertical(newBoard, i, true);
+    }
+    setNewBoard(newBoard);
+  };
+
+  const isGameOver = (array) => {
+    const emptySpots = array.filter((elem) => !elem).length;
+    if (emptySpots) return false;
+    for (let i = 0; i < squareBoardSize; i += boardSize) {
+      const line = getRow(array, i);
+      for (let j = 0; j < line.length - 1; j++) {
+        if (line[j] === line[j + 1]) {
+          return false;
+        }
+      }
+    }
+
+    for (let i = 0; i < boardSize; i++) {
+      const line = getColumn(array, i);
+      for (let j = 0; j < line.length - 1; j++) {
+        if (line[j] === line[j + 1]) {
+          return false;
+        }
+      }
+    }
+
+    return true;
+  };
+
+  const handleEvent = (e) => {
+    if (e.stopPropagation) e.stopPropagation();
+    if (e.code === 'ArrowLeft' || e.code === 'KeyA') moveLeft();
+    if (e.code === 'ArrowRight' || e.code === 'KeyD') moveRight();
+    if (e.code === 'ArrowUp' || e.code === 'KeyW') moveUp();
+    if (e.code === 'ArrowDown' || e.code === 'KeyS') moveDown();
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', handleEvent);
+    return () => {
+      document.removeEventListener('keydown', handleEvent);
+    };
+  }, ['keydown', handleEvent]);
+
+  useEffect(() => {
+    let count = 1;
+    const interval = setInterval(() => {
+      if (play && !gameOver) {
+        moveUp();
+        if (count % 3 === 0) moveRight();
+        if (count % 2 === 0) moveLeft();
+        // if (count % 2 !==0) moveDown();
+        count++;
+      }
+    }, 700);
+    return () => clearInterval(interval);
+  }, [play, gameOver, history]);
+
+  const autoPlay = () => (play ? setPlay(false) : setPlay(true));
+
+  return (
+    <footer className="footer">
+      <div className="footer__info">
+        HOW TO PLAY: Use your arrow keys on keyboard or in app to move the tiles (w,a,s,d works
+        too). Tiles with the same number merge into one when they touch. Add them up to reach 64 or
+        2048(can choose in menu)! You can continue to play after you have reached 64 or 2048. Auto
+        play runs until game over(click play button). Also you can stop auto play clicking again
+        same button. Step back button does step back))
+      </div>
+      <div className="footer__bottom">
+        <div className="footer__links">
+          <a
+            href="https://github.com/I-vasilich-I"
+            className="footer__link"
+            target="_blank"
+            rel="noreferrer"
+          >
+            <img
+              className="github__logo"
+              src="assets/images/GitHub-Mark.png"
```

Comment:
svg is better for performance optimization

Reply:
noted